ecosystem/typescript/sdk/examples/typescript/multisig_transaction.ts

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable no-console */
+import { AptosClient, AptosAccount, FaucetClient, BCS, TxnBuilderTypes, TransactionBuilderMultiEd25519 } from 'aptos';
+
+const NODE_URL = process.env.APTOS_NODE_URL || 'https://fullnode.devnet.aptoslabs.com';
+const FAUCET_URL = process.env.APTOS_FAUCET_URL || 'https://faucet.devnet.aptoslabs.com';
+
+type SigningMessage = TxnBuilderTypes.SigningMessage;
+
+/**
+ * This code example demonstrates the process of moving test coins from one multisig
+ * account to a single signature account.
+ */
+(async () => {
+  const client = new AptosClient(NODE_URL);
+  const faucetClient = new FaucetClient(NODE_URL, FAUCET_URL, null);
+
+  // Genereate 3 key pairs and account instances
+  const account1 = new AptosAccount();
+  const account2 = new AptosAccount();
+  const account3 = new AptosAccount();
+
+  // Create a 2 out of 3 MultiEd25519PublicKey. '2 out of 3' means for a multisig transaction
+  // to be executed, at least 2 accounts must have signed the transaction.
+  // See https://aptos-labs.github.io/ts-sdk-doc/classes/TxnBuilderTypes.MultiEd25519PublicKey.html#constructor
+  const multiSigPublicKey = new TxnBuilderTypes.MultiEd25519PublicKey(
+    [
+      new TxnBuilderTypes.Ed25519PublicKey(account1.signingKey.publicKey),
+      new TxnBuilderTypes.Ed25519PublicKey(account2.signingKey.publicKey),
+      new TxnBuilderTypes.Ed25519PublicKey(account3.signingKey.publicKey),
+    ],
+    // Threshold
+    2,
+  );
+
+  // Each Aptos account stores an auth key. Initial account address can be derived from auth key.
+  // See https://aptos.dev/basics/basics-accounts for more details.
+  const authKey = TxnBuilderTypes.AuthenticationKey.fromMultiEd25519PublicKey(multiSigPublicKey);
+
+  // Derive the multisig account address and fund the address with 5000 TestCoin.
+  const mutisigAccountAddress = authKey.derivedAddress();
+  await faucetClient.fundAccount(mutisigAccountAddress, 5000);
+
+  let resources = await client.getAccountResources(mutisigAccountAddress);
+  let accountResource = resources.find((r) => r.type === '0x1::Coin::CoinStore<0x1::TestCoin::TestCoin>');
+  console.log(`multisig account coins: ${(accountResource.data as any).coin.value}. Should be 5000!`);
```

Comment:
we should use some sort of assert here so that we can actually make these tests too.